psi_cardinality/go/bloomfilter/bloomfilter.go

```diff
@@ -0,0 +1,120 @@
+package bloomfilter
+
+/*
+#cgo LDFLAGS: -L${SRCDIR}/../extern/lib/
+#cgo LDFLAGS: -Wl,-rpath,${SRCDIR}/../extern/lib
+#cgo CFLAGS: -I${SRCDIR}/../../cpp
+#cgo LDFLAGS: -lbloom_filter -lbn_util -lcrypto -lopenssl_init -lstatus -lglog -lgflags -lstrings -linternal -lraw_logging_internal -lthrow_delegate -lint128
+
+#include "bloom_filter_c.h"
+*/
+import "C"
+import (
+	"errors"
+)
+
+//BloomFilter context
+type BloomFilter struct {
+	context C.bloom_filter_ctx
+}
+
+//Create returns a new BloomFilter
+func Create(fpr float64, maxElements int64) (*BloomFilter, error) {
+	bf := new(BloomFilter)
+	bf.context = C.bloom_filter_create(C.double(fpr), C.long(maxElements))
+	if bf.context == nil {
+		return nil, errors.New("failed to init bloom filter")
+	}
+	return bf, nil
+}
+
+//CreateFromJSON returns a new BloomFilter from a JSON
+func CreateFromJSON(encoded string) (*BloomFilter, error) {
+	bf := new(BloomFilter)
+	bf.context = C.bloom_filter_create_from_json(C.CString(encoded))
+	if bf.context == nil {
+		return nil, errors.New("failed to init bloom filter")
+	}
+	return bf, nil
+}
+
+//Destroy the context
+func (b *BloomFilter) Destroy() error {
```

Comment:
Would it be possible to not export Destroy, but register it as a Finalizer using [runtime.SetFinalizer](https://golang.org/pkg/runtime/#SetFinalizer)? We don't need to free external resources (DB, files), only memory, so I think the usual issues with Finalizers (that they might not run) should be fine here?

Reply:
I removed the bloomfilter bindings

Reply:
Fixed for the client/server allocations